frontend/app/routes/protected/person-case/birth-details.tsx

```diff
@@ -0,0 +1,275 @@
+          }),
+          v.object({
+            country: v.pipe(
+              v.string(t('protected:birth-details.country.required-country')),
+              v.nonEmpty(t('protected:birth-details.country.required-country')),
+              v.excludes(canadaCountryCode, t('protected:birth-details.country.invalid-country')),
+              v.picklist(COUNTRIES, t('protected:birth-details.country.invalid-country')),
+            ),
+            province: v.optional(
+              v.pipe(
+                v.string(t('protected:birth-details.province.required-province')),
+                v.trim(),
+                v.nonEmpty(t('protected:birth-details.province.required-province')),
+                v.maxLength(maxStringLength, t('protected:birth-details.province.invalid-province')),
+                v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:birth-details.province.invalid-province')),
+              ),
+            ),
+            city: v.optional(
+              v.pipe(
+                v.string(t('protected:birth-details.city.required-city')),
+                v.trim(),
+                v.nonEmpty(t('protected:birth-details.city.required-city')),
+                v.maxLength(maxStringLength, t('protected:birth-details.city.invalid-city')),
+                v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:birth-details.city.invalid-city')),
+              ),
+            ),
+            fromMultiple: v.boolean(t('protected:birth-details.from-multiple.required-from-multiple')),
+          }),
+        ],
+        t('protected:birth-details.country.required-country'),
+      ) satisfies v.GenericSchema<BirthDetailsSessionData>;
+
+      const input = {
+        country: formData.get('country') as string,
+        province: trimToUndefined(formData.get('province') as string),
+        city: trimToUndefined(formData.get('city') as string),
+        fromMultiple: formData.get('from-multiple') ? formData.get('from-multiple') === REQUIRE_OPTIONS.yes : undefined,
+      } satisfies Partial<v.InferInput<typeof schema>>;
+
+      const parseResult = v.safeParse(schema, input, { lang });
+
+      if (!parseResult.success) {
+        return data({ errors: v.flatten<typeof schema>(parseResult.issues).nested }, { status: 400 });
+      }
+
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.birthDetails = parseResult.output;
+
+      throw i18nRedirect('routes/protected/request.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`, ErrorCodes.UNRECOGNIZED_ACTION);
+    }
+  }
+}
+
+export default function BirthDetails({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const [country, setCountry] = useState<string | undefined>(loaderData.defaultFormValues.firstName);
+  const [fromMultiple, setFromMultiple] = useState<boolean | undefined>(
+    loaderData.defaultFormValues.supportingDocumentsRequired,
+  );
```

Comment:
I don't like the name `fromMultiple` here. I actually had to pull your branch and explore the various files to understand what this variable represents.

I think this should be named something like `multipleBirth` or `isPluralBirth` or ..  🤷 (if you can think of something better, please use it)

---

This comment could also apply to the i18n key used for the input, although it's not as serious there since the actual translated string can be used to infer meaning.